### ISSUE TITLE
Time is not right,the flow is not really finish.

### ISSUE THREAD

**Issue by twolight** [markzhai/init#4]
Future<Boolean> initTask = threadPool.submit(flowTask);

try {
      initTask.get(mTimeout, TimeUnit.MILLISECONDS);
 } catch (Exception e) {
      LogImpl.w(TAG, "timeout for flow: " + getName());
}
long endTime = System.currentTimeMillis();
LogImpl.i(TAG, getName() + " runs " + (endTime - startTime));

Flow is not really finish.Some task may be still running.
And i think flow or Init need a callback,for example in Android loading Activity.

**Comment by markzhai** [comment]
Yep, the time only tells cost for block task, asychronous tasks are not counted in.
